src/common/dsp/oscillators/WindowOscillator.h

```diff
@@ -77,7 +77,7 @@ class WindowOscillator : public Oscillator
     void applyFilter();
     template <bool is_init> void update_lagvals();
 
-    void ProcessWindowOscs(bool stereo, bool FM);
+    template <bool FM, bool Full16> void ProcessWindowOscs(bool stereo);
```

Comment:
Maybe in order to simplify the code, you could convert 16 bits wavetable to float and then have a single wavetable player?

It is a tradeoff, I believe that the win of running the wt on 16 bits is the memory but today I'm not sure how relevant it is versus 1 code path. Especially if you have a GUI.

Reply:
The window oscillator and the wavetable oscillator are really quite different (window convolves with a pre-defined window whereas wavetable does a BLIT chain). Window uses the I16 and wavetable uses the F32 which is why we convert on load. But having a single oscillator would be a breaking change and a feature reduction.

As to why both functions weren't in one oscillator, the pragmatic answer i that the parameter sets are different and parameters used to be relatively non-dynamic. The longer answer is easier for you to get an answer to in your team than it is for me in mine :) (aka : ask claes! )